### PR TITLE
Convert JavaScript sources to TypeScript

### PR DESCRIPTION
**Pull request by strax**
Mostly a compile-time refactor by adding type annotations but some Ramda usages were changed / removed to improve type inference. Didn't adopt TSLint yet because there is no corresponding create-react-app configuration for it at the moment and didn't want to introduce linting changes (although eslint now warns incorrectly about unused variables when imports are used in a type position).

### CODE REVIEW

File: src/components/ShrinkingHeader.tsx [hunk 1]
```diff
@@ -1,10 +1,15 @@
 import React from 'react'
-import * as R from 'ramda'
 import { headerHeight, headerHeightShrunk } from '../common'
-import Header from './Header'
+import Header, { Props as HeaderProps } from './Header'
 
-class ShrinkingHeader extends React.Component {
-  constructor(props) {
+export type Props = Pick<HeaderProps, Exclude<keyof HeaderProps, 'isShrunk'>>
```
Comment (cxcorp) [inline src/components/ShrinkingHeader.tsx:5]: 👌

File: src/components/ShrinkingHeader.tsx [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+
+interface State {
+  isShrunk: boolean
+}
+
+class ShrinkingHeader extends React.Component<Props, State> {
+  constructor(props: Props) {
     super(props)
     this.state = { isShrunk: false }
     this.handleScroll = this.handleScroll.bind(this)
```
Comment (cxcorp) [inline src/components/ShrinkingHeader.tsx:15]: Perhaps we could finally change `handleScroll() {}` to `handleScroll = () => {}` to get rid of this bind

File: src/containers/DummyDocumentList.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import DocumentList from '../components/DocumentList'
+import NotFound from '../components/NotFound'
+import { deserialize } from './deserializer'
+
+const DummyDocumentList = ({
+  match
+}: {
+  match: { params: { courseName: string } }
+}) => {
+  const courses = deserialize(dummyCourses)
```
Comment (cxcorp) [inline src/containers/DummyDocumentList.tsx:12]: 👌 👌 👌

**Review by cxcorp — CHANGES_REQUESTED** [review]

File: src/components/ShrinkingHeader.tsx (hunk 2 above)
Comment (strax) [reply src/components/ShrinkingHeader.tsx:15]: That could be done but i’d avoid scope creep here and keep refactorings to a minimum.

Comment (cxcorp) [reply src/components/ShrinkingHeader.tsx:15]: :+1:

**Pull request merged**